### ISSUE TITLE
Setup artifacts-credprovider in TFS CI/CD with Docker Fails under 5.0

### ISSUE THREAD

**Issue by irperez** [microsoft/artifacts-credprovider#228]
This is in reference to issue #63 

The core issue here is that **dotnet restore** doesn't work with an Azure DevOps (on premise) NuGet feed inside of Docker.  So we need to use a Nuget credential provider to authenticate with TFS so that restore works properly.  

The original was an issue with dotnet SDK 3.1.  That fix no longer works with dotnet SDK 5.0.

Actual error with Azure DevOps on premise:
**GSSAPI operation failed with error - Unspecified GSS failure.  Minor code may provide more information (SPNEGO cannot find mechanisms to negotiate)**

And yes we've been using the above solution for months with 3.1.  MSFT we need you to support Linux here a little better!  This is a brutal solution to the problem.  And the kicker for us here is that 3.1 is NOT LTS.  

Below is the solution that we had working in 3.1 and is now failing under 5.0.

`
FROM mcr.microsoft.com/dotnet/aspnet:5.0-buster-slim AS base
WORKDIR /app
EXPOSE 80
EXPOSE 443

FROM mcr.microsoft.com/dotnet/sdk:5.0-buster-slim AS build
ARG nuget_pat
ARG feed_url

ENV NUGET_CREDENTIALPROVIDER_SESSIONTOKENCACHE_ENABLED true
ENV VSS_NUGET_EXTERNAL_FEED_ENDPOINTS '{"endpointCredentials":[{"endpoint":"$(feed_url)","username":"NoRealUserNameAsIsNotRequired","password":"'${nuget_pat}'"}]}'
ENV DOTNET_SYSTEM_NET_HTTP_USESOCKETSHTTPHANDLER=0 

RUN wget -O - https://raw.githubusercontent.com/Microsoft/artifacts-credprovider/master/helpers/installcredprovider.sh  | bash

WORKDIR /src
COPY . /src/Drugs.API
COPY ["Drugs.API.csproj", "Drugs.API/"]

RUN dotnet restore "Drugs.API/Drugs.API.csproj" -s "${feed_url}"     <==== FAILURE HERE

WORKDIR "/src/Drugs.API"
RUN dotnet build "Drugs.API.csproj" -c Release -o /app/build --no-restore

FROM build AS publish
RUN dotnet publish "Drugs.API.csproj" -c Release -o /app/publish --no-restore

FROM base AS final
WORKDIR /app
COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "Drugs.API.dll"]`

**Comment by irperez** [comment]
So after doing some more homework on this problem, the solution was so simple I had to slap myself silly.   In a nutshell the scripts to get credential provider go away.  Instead its a simple **dotnet nuget add source**.  See the new docker script below.  Hope this helps.

`
FROM mcr.microsoft.com/dotnet/aspnet:5.0-buster-slim AS base
LABEL maintainer="iperez@nanthealth.com"
WORKDIR /app
EXPOSE 80
EXPOSE 443

FROM mcr.microsoft.com/dotnet/sdk:5.0-buster-slim AS build
ARG nuget_pat
ARG feed_url

WORKDIR /src
COPY . /src/Drugs.API
COPY ["Drugs.API.csproj", "Drugs.API/"]

RUN dotnet nuget add source $(feed_url) -u "whatever" -p "${nuget_pat}" --store-password-in-clear-text --valid-authentication-types "basic"
RUN dotnet restore "Drugs.API/Drugs.API.csproj"

WORKDIR "/src/Drugs.API"
RUN dotnet build "Drugs.API.csproj" -c Release -o /app/build --no-restore

FROM build AS publish
RUN dotnet publish "Drugs.API.csproj" -c Release -o /app/publish --no-restore

FROM base AS final
WORKDIR /app
COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "Drugs.API.dll"]
`

**Comment by larserikfinholt** [comment]
ok, I removed the credentials provider, but and added feed like this:

`RUN dotnet nuget add source https://tfsserver/web/_packaging/SmiFeed/nuget/v3/index.json -n SmiFeed -u "dummy" -p "xxxxxxxxxxxxxxxxxxx" --store-password-in-clear-text --valid-authentication-types "basic"`

But we get a nuget error `Unable to load the service index for source` and a 401

How can we debug this? The PAT is correct. I assume the username is not relevant. 

With the credential provider and nuget.config we got `GSSAPI operation failed with error - Unspecified GSS failure.  Minor code may provide more information (SPNEGO cannot find mechanisms to negotiate)`
